sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobSeekableByteChannelTests.groovy

```diff
@@ -0,0 +1,93 @@
+package com.azure.storage.blob.specialized
+
+import com.azure.core.util.BinaryData
+import com.azure.storage.blob.APISpec
+import com.azure.storage.blob.BlobClient
+import com.azure.storage.blob.models.BlobRequestConditions
+import com.azure.storage.blob.models.ConsistentReadControl
+import com.azure.storage.blob.options.BlobSeekableByteChannelReadOptions
+import com.azure.storage.common.StorageSeekableByteChannel
+import com.azure.storage.common.implementation.Constants
+import com.azure.storage.common.test.shared.TestUtility
+
+class BlobSeekableByteChannelTests extends APISpec {
+    BlobClient bc
+
+    def setup() {
+        bc = cc.getBlobClient(generateBlobName())
+    }
+
+    def "E2E channel read"() {
```

Comment:
Should add a live only test to work with >2GB blobs to ensure we are solving the customer issue found in Blob NIO where an exception would happen once we passed the Integer.MAX_VALUE size.

Reply:
I'd rather not subject live tests to more insanely long tests. And with blobs we can't tell the service to make a massive blob and only upload data past the maxint line, we'd have to upload the whole thing. Better imo to mock such a scenario and not hit the network at all.